drivers/wifi/esp_at/esp.c

```diff
@@ -816,9 +816,44 @@ static int cmd_ipd_parse_hdr(struct esp_data *dev,
 		return str - ipd_buf;
 	}
 
+	if (!ESP_PROTO_PASSIVE(esp_socket_ip_proto(*sock)) &&
+	    IS_ENABLED(CONFIG_WIFI_ESP_AT_CIPDINFO_USE)) {
+		struct sockaddr_in *recv_addr =
+			(struct sockaddr_in *) &(*sock)->context->remote;
+		char *remote_ip;
+		long port;
+
+		err = esp_pull_quoted(&str, str_end, &remote_ip);
+		if (err) {
+			LOG_ERR("Failed to pull remote_ip");
+			goto socket_unref;
+		}
+
+		err = esp_pull_long(&str, str_end, &port);
+		if (err) {
+			LOG_ERR("Failed to pull port");
+			goto socket_unref;
+		}
+
+		err = net_addr_pton(AF_INET, remote_ip, &recv_addr->sin_addr);
+		if (err) {
+			LOG_ERR("Invalid IP address");
+			err = -EBADMSG;
+			goto socket_unref;
+		}
+
+		recv_addr->sin_family = AF_INET;
+		recv_addr->sin_port = ntohs(port);
```

Comment:
The sockaddr_in struct expects the port number to be in network byte order, so the `ntohs()` could be omitted.
See https://www.man7.org/linux/man-pages/man3/sockaddr.3type.html for details.

Reply:
Good catch! I have basically reused part of the code from https://github.com/zephyrproject-rtos/zephyr/blob/48d69b4e7e0add3b00320d4568149213ff709867/drivers/wifi/esp_at/esp_offload.c#L547 which handles "passive mode" in the driver.

In both cases `port` is in host (native) byte order, after parsing a string to integer. This means that `htons()` should be used intead of `ntohs()` here. Everything worked so far, since both macros (`htons` and `ntohs`) are actually `BWAP_16()` macro invocations.

I have changed both invocations of `ntohs()` to be `htons()` instead.